docs/spec/sources.md

```diff
@@ -362,5 +376,39 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects).
+
+The `knsource` value SHOULD be of the form `<name>.<plural>.<group>`, where:
```

Comment:
Ideally the adapters we need to use to bridge cloudevents are transparent and don't decorate the event. I am not sure what value it brings to add metadata on how the event occurrence was brought into the cluster. If that data is not understood or worthy of being part of `type`, `source` or `subject does it really have value to the routers and consumers of the event?

Reply:
> Ideally the adapters we need to use to bridge cloudevents are transparent and don't decorate the event

Agree, that's the main reason for the SHOULD.

> I am not sure what value it brings to add metadata on how the event occurrence was brought into the cluster.  If that data is not understood or worthy of being part of `type`, `source` or `subject` does it really have value to the routers and consumers of the event?

The originating CR ref cannot be part of of `type`, `source` or `subject` because their values are usually defined outside of Knative. There are exceptions though (eg. PingSource). It is useful in the context of Knative, and Knative only.

Reply:
I think that is so source specific it makes no sense to add it to the spec.

Reply:
@n3wscott I'm not sure what do you mean by "source specific"? what about the use cases described above? it'd be great if you check them and let us know how else can a knative developer specify a filter when `type`, `source` and `subject` are prepopulated or too broad ? IIRC every knative source can conform very easily to `knsource` cause every source can fill the `<name>.<plural>.<group>` combination, so I'm not sure why is it too source specific?